symplyphysics/conditions/thermodynamics/internal_energy_is_first_order_homogeneous_function.py

```diff
@@ -0,0 +1,56 @@
+from sympy import Eq
+from symplyphysics import (
+    units,
+    dimensionless,
+    Symbol,
+    Function,
+    print_expression,
+)
+from symplyphysics.core.expr_comparisons import expr_equals
+from symplyphysics.laws.thermodynamics import internal_energy_differential as internal_energy_law
+
+# Description
+## Internal energy is a first-order homogeneous function of its internal variables (entropy,
+## volume, and particle count).
+
+# Law: U(k * S, k * V, k * N) = k * U(S, V, N) forall k
+## U = U(S, V, N) - internal energy as a function of its natural variables
+## S - entropy
+## V - volume
+## N - particle count
+## k - any (real) number
+
+internal_energy = Function("internal_energy", units.energy)
+entropy = Symbol("entropy", units.energy / units.temperature)
+volume = Symbol("volume", units.volume)
+particle_count = Symbol("particle_count", dimensionless)
+factor = Symbol("factor", dimensionless)
+
+law = Eq(
```

Comment:
You put it in `conditions` but it looks like a law. You can make a condition from it. Smth like `homogenity_condition`.